main.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/bitrise-io/go-utils/command/git"
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-tools/go-steputils/stepconf"
+)
+
+type config struct {
+	RepositoryURL string `env:"repository_url,required"`
+	CloneIntoDir  string `env:"clone_into_dir,required"`
+	Commit        string `env:"commit"`
+	Tag           string `env:"tag"`
+	Branch        string `env:"branch"`
+}
+
+func mainE() error {
+	var cfg config
+	if err := stepconf.Parse(&cfg); err != nil {
+		failf("Error: %s\n", err)
+	}
+	stepconf.Print(cfg)
+
+	gitCmd, err := git.New(cfg.CloneIntoDir)
+	if err != nil {
+		return fmt.Errorf("create gitCmd project, error: %v", err)
+	}
+	checkoutArg := getCheckoutArg(cfg.Commit, cfg.Tag, cfg.Branch)
```

Comment:
I would remove the getCheckoutArg function and simply pass all the checkout inputs to the checkout function.

The current behavior is a bit strange: getCheckoutArg sets a preferred checkout order (which might try to overcome the case if multiple checkout arg provided): commit -> tag -> branch.

But checkout is not in sync with this preference.

Reply:
i would suggest modifying the logic in the way it works only with 1 checkout arg and fail if multiple provided.